pylint/checkers/exceptions.py

```diff
@@ -337,6 +356,46 @@ def _check_bad_exception_context(self, node):
         ):
             self.add_message("bad-exception-context", node=node)
 
+    def _check_raise_missing_from(self, node: astroid.Raise) -> None:
+        if node.cause is not None:
+            return
+        if node.exc is None:
+            # This is a plain `raise`, raising the previously-caught exception. No need for
+            # a cause.
+            return
+        # We'd like to check whether we're inside an `except` clause:
+        for parent in iterate_parents(node):
+            if isinstance(parent, astroid.ExceptHandler):
```

Comment:
You can put that conditional directly in `iterate_parents` and yield only `astroid.ExceptHandler`. It will remove a level of indentation here which is always nice.

Reply:
Hmm, I don't see that as better. Consider I'll also have to push the class/function check in there.

Do you think that `astroid.NodeNG` should have a `.parents()` method that iterates backwards on parents?

Reply:
Or maybe `.ancestors()`?

Reply:
I don't know if I made much sense I'm thinking of something like this:
````
def iterate_exception(node: NodeNG) -> typing.Iterator[NodeNG]:
    current = node.parent
    while current is not None:
        if isinstance(current, astroid.ExceptHandler):
            yield current
            current = current.parent
````
The class/function check could stay the same I think ?

Reply:
Good thinking, I refactored.